service/history/shard/controller_impl.go

```diff
@@ -353,9 +353,17 @@ func (c *ControllerImpl) acquireShards() {
 			c.contextTaggedLogger.Error("Unable to create history shard context", tag.Error(err), tag.OperationFailed, tag.ShardID(shardID))
 			return
 		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), shardIOTimeout)
+		defer cancel()
+		if err := shard.AssertOwnership(ctx); IsShardOwnershipLostError(err) {
+			c.CloseShardByID(shardID)
```

Comment:
`AssertOwnership` already calls `handleWriteErrorAndUpdateMaxReadLevelLocked`, which handles shard ownership lost errors by initiating shutdown. the error handling should happen in one place. so either this site should only call `AssertOwnership` and trust it to handle errors (my preference) or `AssertOwnership` should not call `handleWriteError...` and all callers should handle errors (this seems worse).